site/en/docs/devtools/javascript/breakpoints/index.md

```diff
@@ -59,37 +60,90 @@ investigate, but you want to pause only when some other condition is true.
 
 To set a conditional line-of-code breakpoint:
 
-1.  Click the **Sources** tab.
+1.  Open the **Sources** tab.
 2.  Open the file containing the line of code you want to break on.
 3.  Go to the line of code.
 4.  To the left of the line of code is the line number column. Right-click it.
 5.  Select **Add conditional breakpoint**. A dialog displays underneath the line of code.
 6.  Enter your condition in the dialog.
-7.  Press Enter to activate the breakpoint. An orange icon appears on top of the line number column.
+7.  Press <kbd>Enter</kbd> to activate the breakpoint. An orange icon with a question mark appears on top of the line number column.
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/wbb7VQOAG1GOHvBqqjdt.png", alt="A conditional line-of-code breakpoint.", width="800", height="579" %}
+
+This example shows a conditional line-of-code breakpoint set on line **30**.
+
+### Log line-of-code breakpoints {: #log-loc }
+
+Use log line-of-code breakpoints (logpoints) to log messages to the **Console** without pausing the execution and without cluttering up your code with `console.log()` calls.
+
+To set a logpoint:
+
+1.  Open the **Sources** tab.
+1.  Open the file containing the line of code you want to break on.
+1.  Go to the line of code.
+1.  To the left of the line of code is the line number column. Right-click it.
+1.  Select **Add logpoint**. A dialog displays underneath the line of code.
+1.  Enter your log message in the dialog.
+    
+    You can separate strings, variable, and objects with commas. Wrap strings in quotes (`" "`), objects in curly brackets (`{ }`), and type variable names as they are.
```

Comment:
> Wrap strings in quotes (`" "`), objects in curly brackets (`{ }`), and type variable names as they are

There are a few thoughts are:

1. Erm... I afraid this might give user wrong impression: this line indicate they must wrap object in curly brackets which is not the case, and string can be wrap in single quotes as well. 
2. I am wondering if we can replace this to something "accept javascript expression"
3. User can also log an expression / condition. For example, here is a few valid logs `"test " + variableName, "test".length > 3, "test".toUpperCase()"`

Given `variableName = 'haha'` This example will log `testhaha, true, TEST` in the console.

I would like to see us expand the section here to cover more use-cases.

Reply:
@ktran WDYT?

Reply:
I agree that it is a bit misleading wrt to objects. I think it's a great idea to add more examples here! 

Actually, using `'x is' + x` is also valid. What happens with log points is that whatever you enter is wrapped into a `console.log`, so maybe adding this info (or something along these lines) will make it more apparent to developers on what syntax they can use here.

Adding @szuend , in case Simon has some input on this.

Reply:
Maybe we can keep the initial instructions relatively simple, as they are now: "You can separate strings, variables, and objects with commas." But add a technical note or a footnote that explains that any JavaScript expression, or comma-separated list of JavaScript expressions is valid since it's wrapped as-is into a `console.log`.

Reply:
I apologise to have hallucinated a misleading description. Thank you for catching that. Fixed.